libparsec/src/config.rs

```diff
@@ -74,3 +76,75 @@ impl From<ClientConfig> for libparsec_client::ClientConfig {
         }
     }
 }
+
+#[derive(Debug, thiserror::Error)]
+pub enum GetServerConfigError {
+    #[error("Cannot communicate with the server: {0}")]
+    Offline(#[from] ConnectionError),
+    #[error(transparent)]
+    Internal(#[from] anyhow::Error),
+}
+
+pub use libparsec_client_connection::protocol::anonymous_server_cmds::latest::server_config::{
+    AccountConfig, ClientAgentConfig, OpenBaoSecretConfig, OrganizationBootstrapConfig,
+};
+
+pub enum OpenBaoAuthConfig {
+    OIDCHexagone { mount_path: String },
+    OIDCProConnect { mount_path: String },
+}
+
+pub struct OpenBaoConfig {
+    pub server_url: String,
+    pub secret: OpenBaoSecretConfig,
+    pub auths: Vec<OpenBaoAuthConfig>,
+}
+
+pub struct ServerConfig {
+    pub client_agent: ClientAgentConfig,
+    pub account: AccountConfig,
+    pub organization_bootstrap: OrganizationBootstrapConfig,
+    pub openbao: Option<OpenBaoConfig>,
+}
+
+pub async fn get_server_config(
+    config_dir: &Path,
+    addr: ParsecAddr,
+) -> Result<ServerConfig, GetServerConfigError> {
+    let cmds = AnonymousServerCmds::new(config_dir, addr, ProxyConfig::default())?;
+
+    use libparsec_client_connection::protocol::anonymous_server_cmds::latest::server_config::{
+        Rep, Req,
+    };
+
+    match cmds.send(Req).await? {
+        Rep::Ok {
+            client_agent,
+            account,
+            organization_bootstrap,
+            openbao,
+        } => Ok(ServerConfig {
+            client_agent,
+            account,
+            organization_bootstrap,
+            openbao: match openbao {
+                libparsec_protocol::anonymous_server_cmds::v5::server_config::OpenBaoConfig::Disabled => None,
+                libparsec_protocol::anonymous_server_cmds::v5::server_config::OpenBaoConfig::Enabled { auths, secret, server_url } => Some(
+                    OpenBaoConfig {
+                        server_url,
+                        secret,
+                        auths: auths.into_iter().filter_map(|auth| match auth.id.as_ref() {
+                            "OIDC_HEXAGONE" => Some(OpenBaoAuthConfig::OIDCHexagone { mount_path: auth.mount_path }),
+                            "OIDC_PRO_CONNECT" => Some(OpenBaoAuthConfig::OIDCProConnect { mount_path: auth.mount_path }),
+                            // Unknown kind of auth, just ignore it
+                            _ => None,
```

Comment:
Maybe add a warning here. IMHO in the best case we support a new auth and the client is late to the update, but worst case the server might be up to something weird.